testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -10857,6 +10857,7 @@ void assertSubFunReturns(boolean binary, String s, int start,
     f.checkFails("trim('' from 'abcde')",
         "Trim error: trim character must be exactly 1 character",
         true);
+    f.checkFails("trim()", "String to trim cannot be null", false);
```

Comment:
The creator of the original https://issues.apache.org/jira/browse/CALCITE-6145 suggested to throw an exception with an exact description.

I've checked the behavior of PostgreSQL (v15):

```sql
SELECT TRIM() AS trimmed_string;
```

leads to a syntax error: `Query Error: error: syntax error at or near ")"`.

```sql
SELECT TRIM("") AS trimmed_string;
```

leads to an error: `Query Error: error: zero-length delimited identifier at or near """"`.

```sql
SELECT TRIM(null) AS trimmed_string;
```

returns `null`.

So I think this aligns with the behavior of PostgreSQL throwing an error during query parsing, _but_ I think the message could be a bit better: `trim cannot be called without arguments` maybe? 

I've added an explicit test for when the value is `null`, which indeed returns `null` (aligns with PostgreSQL's behavior).

WDYT? @mihaibudiu

Reply:
If the function trim() need to have arguments, then the type checker should have complained about not finding a function with a signature without arguments. I don't expect the parser to complain about it, that is strange. The parser should just accept the function, and the type checker should validate it.

The original message is not good, since clearly there is no 'null' involved. However, if the error comes from the parser, there isn't much you can do about it.